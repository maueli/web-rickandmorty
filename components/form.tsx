import React, { useEffect, useState } from 'react';



export const Inputs = ({init, type, dispatch})=>{
    const [ value, setValue ] = useState('');
    useEffect(()=>{
        dispatch({
            type,
            payload:value
        })
    },[value])
    return(
        <div>
            <input name={init} placeholder={init} value={value} type="text" onChange={(e)=>setValue(e.target.value)}/>
        </div>
    )
}

export const Form = ({onSearch, dispatch}) => {
    return(
        <div>
            <Inputs init={'Name'} type='name' dispatch={dispatch}/>
            <Inputs init={'Status'} type='status' dispatch={dispatch}/>
            <Inputs init={'Species'} type='species' dispatch={dispatch}/>
            <Inputs init={'Gender'} type='gender'dispatch={dispatch} />
            <Inputs init={'Origin'} type='origiin' dispatch={dispatch}/>
            <button type='button' onClick={onSearch}>
                Search
            </button>
        </div>
    )
};

