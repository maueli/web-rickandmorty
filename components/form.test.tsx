import renderer from 'react-test-renderer';
import { Inputs } from './form';

it('without blow up',()=>{
    const component = renderer
    .create(<Inputs init='name' type='name' dispatch={()=>{}} />)
    .toJSON()
    expect(component).toMatchSnapshot();
})