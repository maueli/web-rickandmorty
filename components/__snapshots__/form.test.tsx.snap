// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a 1`] = `
<div>
  <input
    name="name"
    onChange={[Function]}
    placeholder="name"
    type="text"
    value=""
  />
</div>
`;

exports[`without blow up 1`] = `
<div>
  <input
    name="name"
    onChange={[Function]}
    placeholder="name"
    type="text"
    value=""
  />
</div>
`;
